### ISSUE TITLE
Release notifications not being received for this repository

### ISSUE THREAD

**Issue by msftrncs** [desktop/desktop#6457]
I have 'watched' this repository for releases, but I have not been getting notifications when releases have been added to this repository.  Is there something more you guys have to do in order to trigger the release notifications?  I get notifications from other repositories when they make releases, both stable and beta.

Thanks

**Comment by j-f1** [comment]
It looks like the releases aren’t turned into GitHub releases but are just left as tags.

**Comment by shiftkey** [comment]
@msftrncs yep, what @j-f1 said - we don't use GitHub releases so you don't get notifications through that mechanism. Our canonical source for releases is somewhere else, so we've not gotten any value from publishing there.

**Comment by shiftkey** [comment]
> I was just hoping to stay on-top of releases, specially on the beta side

This is an interesting idea to explore if people will get value out of it.

We have most of what we need already captured, so I think it's just a matter of understanding the value:

 - version and tag are derived from the source when publishing
 - changelog entries are already set when it's time to publish
 
A couple of concerns I have about also creating a GitHub release each time we do a release:

 - Someone will see the notification, view the release, and then expect to download the latest version from the GitHub release - that's a very different flow to how things are done currently
 - Users being able to access and install older versions isn't something we want to encourage or support - we want them to be on the latest version (for production or beta)

**Comment by shiftkey** [comment]
cc @say25 as I know you have Feelings™ on this stuff

**Comment by msftrncs** [comment]
@shiftkey, it appears that VSCode does the same thing, since they also perform in-app updates.   Maybe the best would just be GitHub notifying that a tag was added, as that's considered a release in this context.

**Comment by j-f1** [comment]
You could do what we do in Prettier and [just put a link to the changelog in the release notes](https://github.com/prettier/prettier/releases). Alternatively, you could create an RSS feed that people could subscribe to to get notifications about releases.

**Comment by say25** [comment]
@shiftkey I think it would be ok for the GitHub Release to lag a touch behind the Production Push release. I did "Release Watch" the repo as well

**Comment by j-f1** [comment]
It would be cool if there was a GitHub action that:
- added a description to newly-created releases based on the `changelog.json` file, and
- updated relevant releases when changes to the `changelog.json` file got merged to `master`

**Comment by NiklasBr** [comment]
I, too, thought that watching for releases would include tagged releases. :)

**Comment by shiftkey** [comment]
> It would be cool if there was a GitHub action that:

Access to GitHub Actions for public repositories is not available currently 😢 

> Alternatively, you could create an RSS feed that people could subscribe to to get notifications about releases.

I like this idea, but I'm not sure how many people are using RSS readers these days. If this is something you'd get value out of, please 👍 so I can gauge interest in exploring this further.

**Comment by j-f1** [comment]
@shiftkey looks like you just have to [register](https://github.com/features/actions/signup/?account=desktop). Actions were [recently enabled](https://developer.github.com/actions/changes/2018-12-18-limited-public-actions/) for public repos when commits are pushed.

**Comment by shiftkey** [comment]
@j-f1 I've registered but I'm not holding my breath for it 😛

**Comment by say25** [comment]
I believe this is now available to be solved via GitHub actions

**Comment by shiftkey** [comment]
There's two goals that I'd like to achieve as part of this process:

 - when a tag matching the release format is pushed (for production or beta), create a release entry with information as well as links about where to find the release
 - add a Markdown-ified changelog to the release based on the release notes

This is on my list once I make a dent into the rebasing work.

We also have the ability to refresh the changelog on the marketing site from `development`, and keeping these releases in sync is that's something I'd like to support but is a lower-priority currently.

**Comment by raiyanalam** [comment]
As mentioned in #7573, we may already have a solution here. Azure pipelines provides an[ inbuilt task](https://github.com/microsoft/azure-pipelines-tasks/tree/master/Tasks/GitHubReleaseV0) to create release in GitHub. It can be configured to create a release only when the pipeline is triggered for a matching tag. Also, it can take the contents of a file that's checked into repository and add it as the release notes. 

Addressing couple of concerns raised by @shiftkey earlier:

> Someone will see the notification, view the release, and then expect to download the latest version from the GitHub release - that's a very different flow to how things are done currently

Many repositories provide a link to an external web page for downloading binaries at the beginning of release notes or sometimes an external link is all they have in release notes. For this repo, a link to download can be prepended to release notes to direct users to the GitHub Desktop webpage.

 
>  Users being able to access and install older versions isn't something we want to encourage or support - we want them to be on the latest version (for production or beta)

With each new release, we can probably delete the older release. This way users get to see only the latest version. Deletion is also supported by the same task.

**Comment by shiftkey** [comment]
@raiyanalam thanks for following up on this, but I feel pulling in Azure Pipelines for this is a bit heavy-handed. I'll walk through what my current thinking on this is to provide more context from my side:

> Azure pipelines provides an[ inbuilt task](https://github.com/microsoft/azure-pipelines-tasks/tree/master/Tasks/GitHubReleaseV0) to create release in GitHub

I was eyeing off GitHub Actions rather than using a CI/CD platform here for three main reasons:

 - the release artifacts are stored and served from a different system, so involving a third party system here feels excessive just to "list the release" (my words)
 - I know enough about the GitHub API (which Actions can consume) to feel that doing this in a Node script will have all the information I need to do the job
 - Because we would have access to anything a normal Node environment would have from inside the Action, we can do whatever custom things we want to generate the release details we need

> Also, it can take the contents of a file that's checked into repository and add it as the release notes.

I'd love to be able to do this for the release body:

 - read the contents of `CHANGELOG.json`
 - extract the array of strings from the JSON associated with the tag that was received
 - format the array of strings into a list of Markdown elements
 - add a paragraph to the end containing download URLs to the release (based on whether this was a `beta` or `production` release)
 - use that as the `body` for the release

That's not too hard to do from inside the Action script once I figure out the format I want (here's an example from our [release notes](https://desktop.github.com/release-notes/) on the main site).

**Comment by drdavi** [comment]
en español

Descarga Outlook para iOS<https://aka.ms/o0ukef>
________________________________
De: Brendan Forster <notifications@github.com>
Enviado: Tuesday, May 28, 2019 5:43:06 AM
Para: desktop/desktop
Cc: Subscribed
Asunto: Re: [desktop/desktop] publishing a release does not generate a GitHub release notification (#6457)


@raiyanalam<https://github.com/raiyanalam> thanks for following up on this, but I feel pulling in Azure Pipelines for this is a bit heavy-handed. I'll walk through what my current thinking on this is to provide more context from my side:

Azure pipelines provides an inbuilt task<https://github.com/microsoft/azure-pipelines-tasks/tree/master/Tasks/GitHubReleaseV0> to create release in GitHub

I was eyeing off GitHub Actions rather than using a CI/CD platform here for three main reasons:

  *   the release artifacts are stored and served from a different system, so involving a third party system here feels excessive just to "list the release" (my words)
  *   I know enough about the GitHub API (which Actions can consume) to feel that doing this in a Node script will have all the information I need to do the job
  *   Because we would have access to anything a normal Node environment would have from inside the Action, we can do whatever custom things we want to generate the release details we need

Also, it can take the contents of a file that's checked into repository and add it as the release notes.

I'd love to be able to do this for the release body:

  *   read the contents of CHANGELOG.json
  *   extract the array of strings from the JSON associated with the tag that was received
  *   format the array of strings into a list of Markdown elements
  *   add a paragraph to the end containing download URLs to the release (based on whether this was a beta or production release)
  *   use that as the body for the release

That's not too hard to do from inside the Action script once I figure out the format I want (here's an example from our release notes<https://desktop.github.com/release-notes/> on the main site).

**Comment by paperh8rrr** [comment]
investor.pypl.com
